File: codes/algorithm/insertion_sort.py
def insertion_sort(arr):
    for j in range(1, len(arr)):
        key = arr[j]
        i = j-1
        while i >= 0 and arr[i] > key: # 부등호 반대하면 내림차순
            arr[i+1] = arr[i]
            i -= 1
            print(arr)
        arr[i+1] = key
    return arr

d = [4,1,2,3,6,5]
print(insertion_sort(d))

File: codes/algorithm/insertion_sort.cpp
#include <iostream>
#include <vector>

using namespace std;

void insertion_sort(vector<int> &arr){
    for(int i = 1; i < arr.size(); i++){
        int key = arr[i];
        int j = i - 1;
        while(j >= 0 && arr[j] > key){
            arr[j+1] = arr[j];
            j -= 1;
        }
        arr[j+1] = key;
    }
}

int main(){
    vector<int> arr = {4,1,2,3,6,5};
    insertion_sort(arr);
    for(auto &e : arr)
        cout << e << " ";
    cout << endl;

    return 0;
}